src/deriving.ml

```diff
@@ -712,8 +716,13 @@ let wrap_sig ~loc ~hide list =
    | Main expansion                                                  |
    +-----------------------------------------------------------------+ *)
 
-let types_used_by_deriving (tds : type_declaration list) : structure_item list =
-  if keep_w32_impl () then []
+let types_used_by_deriving (tds : type_declaration list) ~unused_code_warnings :
+    structure_item list =
+  let unused_code_warnings =
+    allow_unused_code_warnings
+      ~ppx_allows_unused_code_warnings:unused_code_warnings
+  in
+  if keep_w32_impl () || unused_code_warnings then []
```

Comment:
We should probably drop the `keep_w32_impl ()` here, it's clearly a bug to control the unused type warnings via this flag. I'll remove it in a subsequent PR!

Reply:
This sounds good to me. After the set of changes we intend to include in the release, their will be a clear replacement for this (`-unused-type-warnings`). It is not likely to break existing usage, as fixing this bug could only remove warnings rather than adding new ones (right?).

Reply:
Yep that sounds about right!